lib/segment/src/index/field_index/numeric_index/immutable_numeric_index.rs

```diff
@@ -151,13 +150,19 @@ impl<T: Encodable + Numericable> NumericKeySortedVec<T> {
         }
     }
 
-    fn find_end_index(&self, bound: Bound<NumericIndexKey<T>>) -> usize {
+    /// Find the end index of the range.
+    ///
+    /// Scans only starting from `start` to avoid repeated work.
+    fn find_end_index(&self, start: usize, bound: Bound<NumericIndexKey<T>>) -> usize {
         match bound {
-            Bound::Included(bound) => match self.data.binary_search(&bound) {
-                Ok(idx) => idx + 1,
-                Err(idx) => idx,
+            Bound::Included(bound) => match self.data[start..].binary_search(&bound) {
```

Comment:
What if start is `0` and `self.data` is empty? is it possible to meet such a situation? Maybe we have a bug here